plugin/PlayAssetDelivery/src/test/java/com/google/play/core/godot/assetpacks/PlayAssetDeliveryTest.java

```diff
@@ -127,6 +129,114 @@ public void cancel_success() {
     assertThat(resultDict).isEqualTo(testDict);
   }
 
+  @Test
+  public void assetPackStateUpdatedTest() {
+    List<AssetPackState> testAssetPackStateList =
+        PlayAssetDeliveryTestHelper.createAssetPackStateList();
+
+    doAnswer(
+            invocation -> {
+              AssetPackStateUpdateListener listener =
+                  (AssetPackStateUpdateListener) invocation.getArguments()[0];
+              List<AssetPackState> packStateSequence = testAssetPackStateList;
+              for (AssetPackState packState : packStateSequence) {
+                listener.onStateUpdate(packState);
+              }
+              return null;
+            })
+        .when(assetPackManagerMock)
+        .registerListener(any(AssetPackStateUpdateListener.class));
+
+    ArgumentCaptor<String> signalNameCaptor = ArgumentCaptor.forClass(String.class);
+    ArgumentCaptor<Object> signalArgsCaptor = ArgumentCaptor.forClass(Object.class);
+
+    PlayAssetDelivery testSubject = spy(new PlayAssetDelivery(godotMock, assetPackManagerMock));
+
+    testSubject.registerAssetPackStateUpdatedListener();
+
+    verify(testSubject, times(testAssetPackStateList.size()))
+        .emitSignalWrapper(signalNameCaptor.capture(), signalArgsCaptor.capture());
+
+    assertThat(signalNameCaptor.getAllValues())
```

Comment:
I would add a comment here explaining why we expect signalNameCaptor to contain three instances of Asset_PACK_STATE_UPDATED.